fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -2017,16 +2022,24 @@ public void computeColocatedJoinInstanceParam(Map<Integer, TNetworkAddress> buck
                 }
                 List<List<Map.Entry<Integer, Map<Integer, List<TScanRangeParams>>>>> scanRangesPerDriverSeq =
                         ListUtil.splitBySize(scanRangePerInstance, expectedDop);
-                instanceParam.pipelineDop = scanRangesPerDriverSeq.size();
+
+                boolean assignPerDriverSeq = enablePipeline && scanRangesPerDriverSeq.size() > 1;
```

Comment:
Add your PR comment to the code comment to explain why need to check ` scanRangesPerDriverSeq.size() > 1`